sdk/storage/azure-storage-blob/azure/storage/blob/_models.py

```diff
@@ -894,20 +894,39 @@ class ContainerSasPermissions(object):
         List blobs in the container.
     :param bool tag:
         Set or get tags on the blobs in the container.
+    :param bool add:
+        Add a block to any append blob in the container.
+    :param bool create:
+        Write a new blob to the container, snapshot any blob in the container, or copy a blob to
+        a new blob in the container. Note: You cannot grant permissions to create a container
+        with a container SAS. Use an account SAS to create a container instead.
+    :param bool write:
     """
-    def __init__(self, read=False, write=False, delete=False, list=False, delete_previous_version=False, tag=False):  # pylint: disable=redefined-builtin
+    def __init__(self,
+                 read=False,
+                 write=False,
+                 delete=False,
+                 list=False,    # pylint: disable=redefined-builtin
+                 delete_previous_version=False,
+                 tag=False,
+                 add=False,
+                 create=False):  # pylint: disable=redefined-builtin
         self.read = read
         self.write = write
         self.delete = delete
         self.list = list
         self.delete_previous_version = delete_previous_version
         self.tag = tag
+        self.add = add
+        self.create = create
         self._str = (('r' if self.read else '') +
                      ('w' if self.write else '') +
                      ('d' if self.delete else '') +
                      ('x' if self.delete_previous_version else '') +
                      ('l' if self.list else '') +
-                     ('t' if self.tag else ''))
+                     ('t' if self.tag else '') +
+                     ('a' if self.add else '') +
+                     ('c' if self.create else ''))
```

Comment:
the sequence of permission matters, can you double check the position of add and create?